common/readline.c

```diff
@@ -1,51 +1,43 @@
-#include "readline.h"
-#include "log.h"
-#include <stdlib.h>
+#include <assert.h>
 #include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
 
-char *read_line(FILE *file) {
-	size_t length = 0, size = 128;
-	char *string = malloc(size);
-	char lastChar = '\0';
-	if (!string) {
-		sway_log(L_ERROR, "Unable to allocate memory for read_line");
-		return NULL;
-	}
-	while (1) {
-		int c = getc(file);
-		if (c == '\n' && lastChar == '\\'){
-			--length; // Ignore last character.
-			lastChar = '\0';
-			continue;
-		}
-		if (c == EOF || c == '\n' || c == '\0') {
-			break;
-		}
-		if (c == '\r') {
-			continue;
-		}
-		lastChar = c;
-		if (length == size) {
-			char *new_string = realloc(string, size *= 2);
-			if (!new_string) {
-				free(string);
-				sway_log(L_ERROR, "Unable to allocate memory for read_line");
-				return NULL;
+#include "log.h"
+#include "readline.h"
+#include "stringop.h"
+
+/***************************************************************************//**
+*
+*	\brief	Read one line of the file. Lines beginning with '#' are skipped.
+*		Whitespace is stripped.
+*
+*	\param[in]	file	Pointer to the already open configuration file. If NULL is
+*		passed, no operation is performed.
+*
+*	\return		String containing one single line from the file.
+*
+******************************************************************************/
```

Comment:
-1 to all the changes in this file.

- Don't add this sort of doc comments thing. If you want to add doc comments, you should propose it in a seperate pull request. Note that we already use doxygen style comments informally in some other places as well, so if you wanted to do that sort of thing that'd be fine with me, but you should put it in the header.
- Making read_line strip whitespace is beyond what this function should do. If the caller wants whitespace stripped, the caller should strip it themselves.